src/term_mplex.c

```diff
@@ -367,6 +367,10 @@ static void mplex_timer_handler(Toxic *toxic)
         return;
     }
 
+    if (!toxic->c_config->mplex_away) {  // needed in case config is changed during a running session
```

Comment:
`mplex_timer_handler()` is called from a different thread, `mplex_timer_thread`, than the thread that sets `toxic->c_config->mplex_away`. Is it safe to read `toxic->c_config->mplex_away` in here?

Reply:
Hm. Looking further, I see that it's defined as `const char *mplex_away;`, so while it's not an atomic bool, it's kinda smelly but I guess it's fine to check if it's null or non-null without any thread safety in place.

Reply:
Perhaps might make sense to make it volatile, i.e. `volatile const char *mplex_away;`?

Reply:
Ah, I was looking at the wrong `mplex_away`:

https://github.com/TokTok/toxic/blob/2252b80031086efec24dc9f31f7632c713f14a51/src/settings.c#L76

So it is a bool after all:

https://github.com/TokTok/toxic/blob/2252b80031086efec24dc9f31f7632c713f14a51/src/settings.h#L82

Yeah, in that case you need some sort of an atomic bool or locking. Volatile will likely work too, but it's probably a bad idea to use it in this case as it depends on how the compiler implements bool.

Reply:
There's already a mutex that can handle this, I just forgot about it. Good catch.